src/rust/src/decoder/mod.rs

```diff
@@ -86,10 +88,161 @@ impl dtvcc_ctx {
             ),
         }
     }
-    pub fn process_current_packet(&mut self, len: i32) {
+    /// Process current packet into service blocks
+    pub fn process_current_packet(&mut self, len: u8) {
+        let seq = (self.current_packet[0] & 0xC0) >> 6;
+        debug!(
+            "dtvcc_process_current_packet: Sequence: {}, packet length: {}",
+            seq, len
+        );
+        if self.current_packet_length == 0 {
+            return;
+        }
+
+        if self.current_packet_length != len as i32 {
+            // Is this possible?
+            self.reset_decoders();
+            return;
+        }
+
+        if self.last_sequence != CCX_DTVCC_NO_LAST_SEQUENCE
+            && (self.last_sequence + 1) % 4 != seq as i32
+        {
+            warn!("dtvcc_process_current_packet: Unexpected sequence number, it is {} but should be {}", seq, (self.last_sequence +1) % 4);
+        }
+        self.last_sequence = seq as i32;
+
+        let mut pos: u8 = 1;
+        while pos < len {
+            let mut service_number = (self.current_packet[pos as usize] & 0xE0) >> 5; // 3 more significant bits
+            let block_length = self.current_packet[pos as usize] & 0x1F; // 5 less significant bits
+            debug!("dtvcc_process_current_packet: Standard header Service number: {}, Block length: {}", service_number, block_length);
+
+            if service_number == 7 {
+                // There is an extended header
```

Comment:
A reference to the part of the specs that explain this magic number would be good :-)